src/features/import-assertions.md

```diff
@@ -0,0 +1,82 @@
+---
+title: 'Import assertions'
+author: 'Dan Clark ([@dandclark1](https://twitter.com/dandclark1))'
+avatars:
+  - 'dan-clark'
+date: 2021-06-14
+tags:
+  - ECMAScript
+description: 'Import assertions allow module import statements to include additional information alongside the module specifier'
+tweet: ''
+---
+
+The new [import assertions](https://github.com/tc39/proposal-import-assertions) feature allows module import statements to include additional information alongside the module specifier. An initial use for the feature is to enable JSON documents to be imported as [JSON modules](https://github.com/tc39/proposal-json-modules):
+
+```javascript
+// foo.json
+{ "answer": 42 }
+
+// main.mjs
+import json from './foo.json' assert { type: 'json' };
+console.log(json.answer); // 42
+```
+
+## Background: JSON modules and MIME type
+
+A natural question to ask is why a JSON module couldn't simply be imported like this:
+
+```javascript
+import json from './foo.json';
+```
+
+The web platform checks the MIME type of a module resource for validity prior to executing it, and in theory this MIME type could also be used to determine whether to treat the resource as a JSON or as a JavaScript module.
+
+But, there's a [security issue](https://github.com/w3c/webcomponents/issues/839) with relying on the MIME type alone.
+
+Modules can be imported cross-origin, and a developer might import a JSON module from a third-party source. They might consider this to be basically safe even from an untrusted third-party as long as the JSON is properly sanitized, since importing JSON won't execute script.
+
+However, third-party script can actually execute in this scenario because the third-party server could unexpectedly reply with a JavaScript MIME type and a malicious JavaScript payload, running code in the importer's domain.
+
+```javascript
+// Executes JS if sketchy-third-party.com responds with
```

Comment:
I've noticed this causes scrollbars to appear on most devices for import statement. How about we use `evil.com` instead? (it's commonly used in CSP examples, so seems appropriate here too)

Reply:
@danelphick Ping - what do you think? I guess we'll publish this tomorrow then, but let's flesh out those final details before merging.

Reply:
I agree with this. I just pushed a change switching it to evil.com and it looks like this gets rid of the scrollbars, at least on my device.

Reply:
Oh, huh, Github didn't load the new commits and the conversation was unresolved too so I assumed these were still open. In terms of timing, I still think it's best if we publish tomorrow now though, but this looks good, thanks.